File: src/main/java/nl/rabobank/assessment/persistence/repository/PersonRepository.java
package nl.rabobank.assessment.persistence.repository;

import nl.rabobank.assessment.persistence.entity.Person;
import org.springframework.data.repository.reactive.ReactiveCrudRepository;
import reactor.core.publisher.Mono;

import java.util.UUID;

public interface PersonRepository extends ReactiveCrudRepository<Person, UUID> {

	Mono<Boolean> existsByFirstNameAndLastName(String firstName, String lastName);

	Mono<Person> findFirstByFirstName(String firstName);

	Mono<Person> findPersonByFirstNameAndLastName(String firstName, String lastName);

	Mono<Person> findFirstByLastName(String lastName);
}

File: src/main/kotlin/nl/rabobank/assessment/persistence/repository/PersonRepository.kt
package nl.rabobank.assessment.persistence.repository

import nl.rabobank.assessment.persistence.entity.Person
import org.springframework.data.repository.reactive.ReactiveCrudRepository
import reactor.core.publisher.Mono
import java.util.*

interface PersonRepository : ReactiveCrudRepository<Person, UUID> {

    fun existsByFirstNameAndLastName(firstName: String, lastName: String): Mono<Boolean>

    fun findFirstByFirstName(firstName: String): Mono<Person>

    fun findPersonByFirstNameAndLastName(firstName: String, lastName: String): Mono<Person>

    fun findFirstByLastName(lastName: String): Mono<Person>
}